### PR TITLE
Implements #356 (b) Added pageLoadingValidators and loading checks in click methods

### PR DESCRIPTION
**Pull request by ibugaienko**
- [X] By placing an `X` in the preceding checkbox, I verify that I am a PayPal employee or
I have signed the [Contributor License Agreement](https://github.com/paypal/SeLion#contributing)

---
This pull request addresses problem discussed in #356 (2nd solution)

### CODE REVIEW

**Comment by mach6** [comment]
@ibugaienko code aside (the code looks good), It's not clear to me what the difference is between a `pageValidator` and `pageLoadingValidator`. When would a user use on over the other?

**Comment by mach6** [comment]
@ibugaienko After looking closer at the code and the re-reading #356, it seems your intent is to have `pageLoadingValidators` process before any `pageValidators` does. Furthermore, like `pageValidators`, `pageLoadingValidators` appear to be completely optional. :)

Do you think `pageLoadingValidators` should also support negation (like your `pageValidators` PR)?

File: client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java [hunk 1]
```diff
@@ -485,6 +485,9 @@ public void click(Object... expected) {
         if (expected == null || expected.length == 0) {
             return;
         }
+        if (parent != null) {
+            WebDriverWaitUtils.waitUntilPageIsLoaded(parent.getCurrentPage());
+        }
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java:490]: Re: lines 510 to 512 in this this file, when `expected` contains a `WebPage` do you need to `waitUntilPageIsLoaded` on the `WegPage` before calling `waitUntilPageIsValidated`?

**Comment by ibugaienko** [comment]
@mach6 , thank you for feedback!
You are right, in essence, `pageValidators` and `pageLoadingValidators` are same things, that is they are checklists pertaining to a particular page context, and both are optional. Former are used to identify pages, when all are `true`, latter indicate that page is loaded, when all are `false`.

To be honest, I can not think of a good example where negation would be needed in this case, but it could be added quite easily.

File: client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java (hunk 1 above)
Comment (ibugaienko) [reply client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java:490]: This is a very good question, I think you're right! In this case, loading check might be excessive. Should I move `waitUntilPageIsLoaded` inside the for loop and apply to first three if statements ?

**Comment by mach6** [comment]
@ibugaienko  any update on this?

**Comment by ibugaienko** [comment]
@mach6 , Hi, sorry for delay. I thought about your suggestion, but left code as it is. Here is why: there is no need to do loading checks if expected contains *only* another webpage, but if it is not the only expected object then the validation is still required, isn't it?

**Comment by mach6** [comment]
@ibugaienko this is merged. Thanks!

**Pull request closed without merging**